src/pages/iou/MoneyRequestModal.js

```diff
@@ -0,0 +1,449 @@
+        error: PropTypes.bool,
+
+        // Selected Currency Code of the current request
+        selectedCurrencyCode: PropTypes.string,
+    }),
+
+    /** Personal details of all the users */
+    personalDetails: PropTypes.shape({
+        /** Primary login of participant */
+        login: PropTypes.string,
+
+        /** Display Name of participant */
+        displayName: PropTypes.string,
+
+        /** Avatar url of participant */
+        avatar: PropTypes.string,
+    }),
+
+    /** Personal details of the current user */
+    currentUserPersonalDetails: PropTypes.shape({
+        // Local Currency Code of the current user
+        localCurrencyCode: PropTypes.string,
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    hasMultipleParticipants: false,
+    report: {
+        participants: [],
+    },
+    iouType: CONST.IOU.IOU_TYPE.REQUEST,
+    currentUserPersonalDetails: {
+        localCurrencyCode: CONST.CURRENCY.USD,
+    },
+    personalDetails: {},
+    iou: {
+        creatingIOUTransaction: false,
+        error: false,
+        selectedCurrencyCode: null,
+    },
+};
+
+// Determines type of step to display within Modal, value provides the title for that page.
+const Steps = {
+    IOUAmount: 'iou.amount',
+    IOUParticipants: 'iou.participants',
+    IOUConfirm: 'iou.confirm',
+};
+
+const MoneyRequestModal = (props) => {
+    const reportParticipants = lodashGet(props, 'report.participants', []);
+    const participantsWithDetails = _.map(OptionsListUtils.getPersonalDetailsForLogins(reportParticipants, props.personalDetails), personalDetails => ({
+        login: personalDetails.login,
+        text: personalDetails.displayName,
+        firstName: lodashGet(personalDetails, 'firstName', ''),
+        lastName: lodashGet(personalDetails, 'lastName', ''),
+        alternateText: Str.isSMSLogin(personalDetails.login) ? Str.removeSMSDomain(personalDetails.login) : personalDetails.login,
+        icons: [{
+            source: ReportUtils.getAvatar(personalDetails.avatar, personalDetails.login),
+            name: personalDetails.login,
+            type: CONST.ICON_TYPE_AVATAR,
+        }],
+        keyForList: personalDetails.login,
+        payPalMeAddress: lodashGet(personalDetails, 'payPalMeAddress', ''),
+        phoneNumber: lodashGet(personalDetails, 'phoneNumber', ''),
+    }));
+
+    // Skip IOUParticipants step if participants are passed in
+    const steps = reportParticipants.length ? [Steps.IOUAmount, Steps.IOUConfirm] : [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm];
+
+    const prevCreatingIOUTransactionStatusRef = useRef(lodashGet(props.iou, 'creatingIOUTransaction'));
+    const prevNetworkStatusRef = useRef(props.network.isOffline);
+
+    const [previousStepIndex, setPreviousStepIndex] = useState(0);
+    const [currentStepIndex, setCurrentStepIndex] = useState(0);
+    const [participants, setParticipants] = useState(participantsWithDetails);
```

Comment:
Another possible follow up, but we should definitely use a lazy init for `useState()` here.. we need to pull the logic from line 109 down into the `useState()` call so we only calculate it once instead of on every render (might not be hurting anything but it's unnecessary work)...

```suggestion
    // Skip IOUParticipants step if participants are passed in
    const steps = reportParticipants.length ? [Steps.IOUAmount, Steps.IOUConfirm] : [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm];

    const prevCreatingIOUTransactionStatusRef = useRef(lodashGet(props.iou, 'creatingIOUTransaction'));
    const prevNetworkStatusRef = useRef(props.network.isOffline);

    const [previousStepIndex, setPreviousStepIndex] = useState(0);
    const [currentStepIndex, setCurrentStepIndex] = useState(0);
    const [participants, setParticipants] = useState(() => _.map(OptionsListUtils.getPersonalDetailsForLogins(reportParticipants, props.personalDetails), personalDetails => ({
        login: personalDetails.login,
        text: personalDetails.displayName,
        firstName: lodashGet(personalDetails, 'firstName', ''),
        lastName: lodashGet(personalDetails, 'lastName', ''),
        alternateText: Str.isSMSLogin(personalDetails.login) ? Str.removeSMSDomain(personalDetails.login) : personalDetails.login,
        icons: [{
            source: ReportUtils.getAvatar(personalDetails.avatar, personalDetails.login),
            name: personalDetails.login,
            type: CONST.ICON_TYPE_AVATAR,
        }],
        keyForList: personalDetails.login,
        payPalMeAddress: lodashGet(personalDetails, 'payPalMeAddress', ''),
        phoneNumber: lodashGet(personalDetails, 'phoneNumber', ''),
    })));
```

Reply:
Good point, will add that to the list of things to do in https://github.com/Expensify/App/issues/16942